src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -23,24 +23,29 @@ final case class SdkException(
     uri: Option[Uri] = None,
     requestId: Option[String] = None,
     responseCode: Option[Int] = None
-) extends Throwable(message) {
-  override def toString: String = {
+) extends Throwable(SdkException.formatMessage(message, uri, requestId, responseCode))
+object SdkException {
```

Comment:
`toString` on the throwable is used in lots of places, for example ammonite prints this, which is what we want:

```scala
@ import $ivy.`com.cognite::cognite-sdk-scala:1.1.2`; import com.cognite.sdk.scala.v1._
import $ivy.$                                     ;
import com.cognite.sdk.scala.v1._
@ val client = Client("scala-sdk-test", "api.cognitedata.com")
client: Client = Client("scala-sdk-test", "api.cognitedata.com")
@ client.login.status()
com.cognite.sdk.scala.common.SdkException: Unexpected status code 301, in response (with response code 301) to request sent to http://api.cognitedata.com/login/status (with response code 301)
  com.cognite.sdk.scala.v1.RequestSession.unsafeBody(Client.scala:85)
  com.cognite.sdk.scala.v1.RequestSession.$anonfun$get$1(Client.scala:107)
  cats.package$$anon$1.map(package.scala:48)
  cats.Functor$Ops.map(Functor.scala:12)
  cats.Functor$Ops.map$(Functor.scala:12)
  cats.Functor$ToFunctorOps$$anon$4.map(Functor.scala:12)
  com.cognite.sdk.scala.v1.RequestSession.get(Client.scala:107)
  com.cognite.sdk.scala.common.Login.status(login.scala:19)
  ammonite.$sess.cmd15$.<init>(cmd15.sc:1)
  ammonite.$sess.cmd15$.<clinit>(cmd15.sc)
```

Are you sure you didn't catch the exception or did something else in ScalaTest?

Reply:
> Are you sure you didn't catch the exception or did something else in ScalaTest?

I'm quite sure as I did not change anything in these tests except for failing the requests. For example when I checkout master and run

```fish
set -x TEST_API_KEY_READ nokey
sbt test
```

I get (shortened :smile:)

```
[info] com.cognite.sdk.scala.v1.SequenceRowsTest *** ABORTED ***
[info]   com.cognite.sdk.scala.common.InvalidAuthentication: Invalid authentication
[info]   at com.cognite.sdk.scala.v1.GenericClient$.$anonfun$forAuth$2(Client.scala:229)
[info]   at cats.package$$anon$1.map(package.scala:48)
[info]   at cats.Functor$Ops.map(Functor.scala:12)
[info]   at cats.Functor$Ops.map$(Functor.scala:12)
[info]   at cats.Functor$ToFunctorOps$$anon$4.map(Functor.scala:12)
[info]   at com.cognite.sdk.scala.v1.GenericClient$.forAuth(Client.scala:225)
[info]   at com.cognite.sdk.scala.common.SdkTestSpec.<init>(SdkTestSpec.scala:33)
[info]   at com.cognite.sdk.scala.v1.SequenceRowsTest.<init>(SequenceRowsTest.scala:8)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   ...

```

(apart from the message, it's also annoying how it shortens the stack traces. If there is a better way to run the tests, I'd be quite glad)

Reply:
That seems to be a different exception type though. What happens if you (for example) try to fetch an asset that does not exist?

Reply:
Thats right, sorry for that. Non existing files are treated as `CdpApiException` (which does build the message instead of overriding `toString`, btw). When I point the tests to a broken proxy I get this:

```
[info] com.cognite.sdk.scala.v1.SecurityCategoriesTest *** ABORTED ***
[info]   com.cognite.sdk.scala.common.SdkException: Unexpected status code 301
[info]   at com.cognite.sdk.scala.v1.RequestSession.unsafeBody(Client.scala:85)
[info]   at com.cognite.sdk.scala.v1.RequestSession.$anonfun$get$1(Client.scala:107)
[info]   at cats.package$$anon$1.map(package.scala:48)
[info]   at cats.Functor$Ops.map(Functor.scala:12)
[info]   at cats.Functor$Ops.map$(Functor.scala:12)
[info]   at cats.Functor$ToFunctorOps$$anon$4.map(Functor.scala:12)
[info]   at com.cognite.sdk.scala.v1.RequestSession.get(Client.scala:107)
[info]   at com.cognite.sdk.scala.common.Login.status(login.scala:19)
[info]   at com.cognite.sdk.scala.v1.GenericClient$.forAuth(Client.scala:225)
[info]   at com.cognite.sdk.scala.common.SdkTestSpec.<init>(SdkTestSpec.scala:33)
```

from https://github.com/cognitedata/cognite-sdk-scala/blob/master/src/main/scala/com/cognite/sdk/scala/v1/Client.scala#L82, so it should also contain the URL and so on.